src/lib.rs

```diff
@@ -1,287 +1,340 @@
-    /// * This method cannot be called if the spinner is already stopped.
-    pub fn warn(self, msg: StringLiteral) {
-        self.clear();
-        println!("{} {}", init_color(Some("yellow"), "⚠".to_string()), &msg);
+    pub fn warn(mut self, msg: StringLiteral) {
+        self.stop_spinner_thread();
+
+        println!(
+            "{} {}",
+            init_color(Some(Color::Yellow), "⚠".to_string()),
+            &msg
+        );
     }
 
     /// Deletes the last line of the terminal and prints a new spinner.
     ///
     /// # Example
-    /// ```rust
-    /// use spinoff::Spinners;
-    /// use std::thread::sleep;
-    /// use std::time::Duration;
-    ///     
-    /// let mut sp = spinoff::new(Spinners::Dots, "Hello", None);
+    ///
+    /// ```
+    /// # use spinoff::*;
+    /// # use std::thread::sleep;
+    /// # use std::time::Duration;
+    /// #   
+    /// let mut sp = Spinner::new(Spinners::Dots, "Hello", None);
+    ///
     /// sleep(Duration::from_millis(800));
-    /// sp = sp.update(Spinners::Dots2, "Goodbye".into(), None);
+    /// sp.update(Spinners::Dots2, "Goodbye", None);
     /// sleep(Duration::from_millis(800));
+    ///
     /// sp.stop();
     /// ```
     ///
-    /// # Notes
-    /// * This method will delete the last line of the terminal, so it is recommended to not print anything in between the spinner and the new spinner instance.
-    /// * This method cannot be called if the spinner is already stopped.
-    ///
-    pub fn update(
-        self,
-        spinner: Spinners,
-        msg: StringLiteral,
-        color: Option<StringLiteral>,
-    ) -> Spinner {
-        self.still_spinning
-            .store(false, std::sync::atomic::Ordering::Relaxed);
-        delete_last_line(self.msg);
-        self::new(spinner, msg, color)
+    pub fn update<T>(&mut self, spinner: Spinners, msg: T, color: Option<Color>)
+    where
+        T: Into<Cow<'static, str>>,
+    {
+        self.stop_spinner_thread();
+
+        let _ = std::mem::replace(self, Self::new(spinner, msg, color));
     }
+
     /// Deletes the last line of the terminal.
     ///     
     /// # Example
-    /// ```rust
-    /// use spinoff::Spinners;
-    /// use std::thread::sleep;
-    /// use std::time::Duration;
     ///
-    /// let mut sp = spinoff::new(Spinners::Dots, "Hello", None);
+    /// ```
+    /// # use spinoff::{Spinners, Spinner};
+    /// # use std::thread::sleep;
+    /// # use std::time::Duration;
+    /// #
+    /// let mut sp = Spinner::new(Spinners::Dots, "Hello", None);
     /// sleep(Duration::from_millis(800));
     /// sp.clear();
     /// ```
     ///
-    /// # Notes
-    /// * This method will delete the last line of the terminal, so it is recommended to not print anything in between the spinner and the `delete` method call.
-    /// * This method cannot be called if the spinner is already stopped.
-    pub fn clear(self) {
+    pub fn clear(mut self) {
+        self.stop_spinner_thread();
```

Comment:
This is quite confusing since clearing should delete the spinner and message, this only stops the thread.